# load_ext sql is used to prepare the sql environment
## it stands for load extension sql

In [1]:
%load_ext sql

## The syntax to connect to mysql workbench which is an example of DataBase Mangement System is
### %sql mysql+pymsql://{username}:{password}@{Host}:{port}/{database}
#### Database management system is a software that allow user to create, store, retrieve and run queries on data stored in a database
#### Act as interface between the end user and the database

In [2]:
%sql mysql+pymysql://root:mysql2026@127.0.0.1:3306/md_water_services

# The code below is used to find all table names which will be used for queries.

#### when you access a new database in MySQL, a handy initial query to run is SHOW TABLES. This will give you a list of all the tables in the database.

In [3]:
%%sql
SHOW TABLES;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
10 rows affected.


Tables_in_md_water_services
auditor_report
data_dictionary
employee
global_water_access
incorrect_record
location
visits
water_quality
water_source
well_pollution


## After running I have found that they are 8 tables name
### These are data_dictionary, employee, visits, water_quality, water_source, well_pollution, location and global_water_access

# Part 1 – Beginning my data-driven journey in Maji Ndogo

# Context
#### In this first part of the integrated project, we dive into Maji Ndogo's expansive database containing 60,000 records
#### spread across various tables. As we navigate this trove of data, we'll use basic queries to familiarise ourselves with
#### the content of each table. Along the way, we'll also refine some data using DML.

# Learning objective and outcomes
#### Demonstrate the ability to create and navigate through a new database.
#### Demonstrate the ability to create and modify basic queries.
#### Demonstrate the ability to use operators to create conditional filters.

### So let's have a look at one of these tables, Let's use location so we can use that killer query, SELECT * but remember to limit it and tell
### it which table we are looking at.

In [4]:
%%sql
SELECT * FROM md_water_services.location LIMIT 5

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


location_id,address,province_name,town_name,location_type
AkHa00000,2 Addis Ababa Road,Akatsi,Harare,Urban
AkHa00001,10 Addis Ababa Road,Akatsi,Harare,Urban
AkHa00002,9 Addis Ababa Road,Akatsi,Harare,Urban
AkHa00003,139 Addis Ababa Road,Akatsi,Harare,Urban
AkHa00004,17 Addis Ababa Road,Akatsi,Harare,Urban


#### So we can see that this table has information on a specific location, with an address, the province and town the location is in, and if it's
#### in a city (Urban) or not. We can't really see what location this is but we can see some sort of identifying number of that location.

In [5]:
%%sql
SELECT * FROM md_water_services.visits LIMIT 5

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


record_id,location_id,source_id,time_of_record,visit_count,time_in_queue,assigned_employee_id
0,SoIl32582,SoIl32582224,2021-01-01 09:10:00,1,15,12
1,KiRu28935,KiRu28935224,2021-01-01 09:17:00,1,0,46
2,HaRu19752,HaRu19752224,2021-01-01 09:36:00,1,62,40
3,AkLu01628,AkLu01628224,2021-01-01 09:53:00,1,0,1
4,AkRu03357,AkRu03357224,2021-01-01 10:11:00,1,28,14


#### so this is a list of location_id, source_id, record_id, and a date and time, so it makes sense that someone (assigned_em-
#### ployee_id) visited some location (location_id) at some time (time_of_record ) and found a 'source' there (source_id). Often the
#### "_id" columns are related to another table. In this case, the source_id in the visits table refers to source_id in the water_source
#### table. This is what we call a foreign key,

In [6]:
%%sql
SELECT * FROM md_water_services.water_source LIMIT 5

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


source_id,type_of_water_source,number_of_people_served
AkHa00000224,tap_in_home,956
AkHa00001224,tap_in_home_broken,930
AkHa00002224,tap_in_home_broken,486
AkHa00003224,well,364
AkHa00004224,tap_in_home_broken,942


#### Water sources are where people get their water from!

In [7]:
%%sql
SELECT DISTINCT(type_of_water_source) FROM md_water_services.water_source 

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


type_of_water_source
tap_in_home
tap_in_home_broken
well
shared_tap
river


In [8]:
%%sql
SELECT COUNT(*) FROM md_water_services.visits WHERE time_in_queue >= 500 

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
1 rows affected.


COUNT(*)
107


In [9]:
%%sql
SELECT * FROM md_water_services.water_source WHERE source_id IN ("AkRu05234224", "HaZa21742224", "SoRu38776224","AkLu01628224")

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
4 rows affected.


source_id,type_of_water_source,number_of_people_served
AkLu01628224,well,210
AkRu05234224,tap_in_home_broken,496
HaZa21742224,well,308
SoRu38776224,shared_tap,3180


In [10]:
%%sql
SELECT COUNT(*) FROM md_water_services.water_quality WHERE subjective_quality_score >=10 AND visit_count = 2 

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
1 rows affected.


COUNT(*)
218


In [11]:
%%sql
SELECT * FROM md_water_services.well_pollution LIMIT 5

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


source_id,date,description,pollutant_ppm,biological,results
KiRu28935224,2021-01-04 09:17:00,Bacteria: Giardia Lamblia,0.0,495.898,Contaminated: Biological
AkLu01628224,2021-01-04 09:53:00,Bacteria: E. coli,0.0,6.09608,Contaminated: Biological
HaZa21742224,2021-01-04 10:37:00,"Inorganic contaminants: Zinc, Zinc, Lead, Cadmium",2.715,0.0,Contaminated: Chemical
HaRu19725224,2021-01-04 11:04:00,Clean,0.0288593,9.56996e-05,Clean
SoRu35703224,2021-01-04 11:29:00,Bacteria: E. coli,0.0,22.5009,Contaminated: Biological


In [12]:
%%sql
SELECT COUNT(*) FROM md_water_services.well_pollution WHERE results = 'Clean' AND biological > 0.01

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
1 rows affected.


COUNT(*)
64


In [13]:
%%sql
SELECT COUNT(*) FROM md_water_services.well_pollution WHERE description LIKE "Clean%" AND biological > 0.01 ;


 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
1 rows affected.


COUNT(*)
0


In [14]:
%%sql
UPDATE well_pollution 
SET
description = 'Bacteria: E. coli'
WHERE
description = 'Clean Bacteria: E. coli';

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
0 rows affected.


[]

In [15]:
%%sql
UPDATE well_pollution
SET description = "Bacteria: Giardia Lamblia"
WHERE description = 'Clean Bacteria: Giardia Lamblia';

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
0 rows affected.


[]

In [16]:
%%sql
CREATE TABLE md_water_services.well_pollution_copy AS (SELECT * FROM md_water_services.well_pollution );
UPDATE well_pollution_copy
SET description = 'Bacteria: E. coli'
WHERE description = 'Clean Bacteria: E. coli';
UPDATE well_pollution_copy
SET description = "Bacteria: Giardia Lamblia"
WHERE description = 'Clean Bacteria: Giardia Lamblia';
UPDATE well_pollution_copy
SET results = 'Contaminated: Biological'
WHERE biological > 0.01 AND results = 'Clean';

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
17383 rows affected.
0 rows affected.
0 rows affected.
64 rows affected.


[]

In [17]:
%%sql
SELECT * FROM well_pollution_copy WHERE description LIKE "Clean_%" OR (results = "Clean" AND biological > 0.01);

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
0 rows affected.


source_id,date,description,pollutant_ppm,biological,results


In [18]:
%%sql
SELECT COUNT(*) FROM well_pollution_copy WHERE results like 'Clean%' AND biological > 0.01

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
1 rows affected.


COUNT(*)
0


In [19]:
%%sql
DROP TABLE
md_water_services.well_pollution_copy 

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
0 rows affected.


[]

In [20]:
%%sql
SHOW TABLES

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
10 rows affected.


Tables_in_md_water_services
auditor_report
data_dictionary
employee
global_water_access
incorrect_record
location
visits
water_quality
water_source
well_pollution


### I have replace the empty space with a full stop 

In [21]:
%%sql
SELECT 
REPLACE (employee_name, '', '.') AS replace_empty_with_full_stop
FROM employee
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


replace_empty_with_full_stop
Amara Jengo
Bello Azibo
Bakari Iniko
Malachi Mavuso
Cheche Buhle


## lower cased the employee_name

In [22]:
%%sql
SELECT 
LOWER(REPLACE (employee_name, '', '.')) AS Lower_cased_employee_name
FROM employee
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


Lower_cased_employee_name
amara jengo
bello azibo
bakari iniko
malachi mavuso
cheche buhle


### created A new email in the employee table

In [23]:
%%sql
SELECT
CONCAT(REPLACE (employee_name, '', '.') , '@ndogowater.gov') AS new_email
FROM Employee
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


new_email
Amara Jengo@ndogowater.gov
Bello Azibo@ndogowater.gov
Bakari Iniko@ndogowater.gov
Malachi Mavuso@ndogowater.gov
Cheche Buhle@ndogowater.gov


In [24]:
%%sql
UPDATE Employee
SET email = CONCAT(LOWER(REPLACE(employee_name, '', '.')) , '@ndogowater.gov') 

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
56 rows affected.


[]

In [25]:
%%sql
SELECT * FROM Employee LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


assigned_employee_id,employee_name,phone_number,email,address,province_name,town_name,position
0,Amara Jengo,+99637993287,amara jengo@ndogowater.gov,36 Pwani Mchangani Road,Sokoto,Ilanga,Field Surveyor
1,Bello Azibo,+99643864786,bello azibo@ndogowater.gov,129 Ziwa La Kioo Road,Kilimani,Rural,Field Surveyor
2,Bakari Iniko,+99222599041,bakari iniko@ndogowater.gov,18 Mlima Tazama Avenue,Hawassa,Rural,Field Surveyor
3,Malachi Mavuso,+99945849900,malachi mavuso@ndogowater.gov,100 Mogadishu Road,Akatsi,Lusaka,Field Surveyor
4,Cheche Buhle,+99381679640,cheche buhle@ndogowater.gov,1 Savanna Street,Akatsi,Rural,Field Surveyor


In [26]:
%%sql
SELECT 
LENGTH(phone_number)
FROM Employee
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


LENGTH(phone_number)
12
12
12
12
12


In [27]:
%%sql
SELECT
TRIM(phone_number),
LENGTH(TRIM(phone_number))
FROM employee
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


TRIM(phone_number),LENGTH(TRIM(phone_number))
+99637993287,12
+99643864786,12
+99222599041,12
+99945849900,12
+99381679640,12


In [28]:
%%sql
UPDATE employee
SET phone_number = TRIM(phone_number)

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
56 rows affected.


[]

In [29]:
%%sql
SELECT
LENGTH(phone_number)
FROM employee
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


LENGTH(phone_number)
12
12
12
12
12


#### Use the employee table to count how many of our employees live in each town. Think carefully about what function we should use and how we
#### should aggregate the data.

In [30]:
%%sql
SELECT
    town_name,
    COUNT(*) AS num_employees
FROM
    employee
GROUP BY
    town_name;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
9 rows affected.


town_name,num_employees
Ilanga,3
Rural,29
Lusaka,4
Zanzibar,4
Dahabu,6
Kintampo,1
Harare,5
Yaounde,1
Serowe,3


In [31]:
%%sql
SELECT assigned_employee_id, COUNT(*) AS number_of_visits FROM visits 
GROUP BY assigned_employee_id
ORDER BY number_of_visits DESC
LIMIT 3;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
3 rows affected.


assigned_employee_id,number_of_visits
1,3708
30,3676
34,3539


In [32]:
%%sql
SELECT town_name, COUNT(*) AS records_per_town FROM location 
GROUP BY town_name
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


town_name,records_per_town
Harare,1650
Kintampo,780
Lusaka,1070
Rural,23740
Abidjan,400


In [33]:
%%sql
SELECT province_name, COUNT(*) AS records_per_town FROM location 
GROUP BY province_name
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


province_name,records_per_town
Akatsi,8940
Amanzi,6950
Hawassa,6030
Kilimani,9510
Sokoto,8220


#### Can you find a way to do the following:
#### 1. Create a result set showing:
#### • province_name
#### • town_name
#### • An aggregated count of records for each town (consider naming this records_per_town).
#### • Ensure your data is grouped by both province_name and town_name.
#### 2. Order your results primarily by province_name. Within each province, further sort the towns by their record counts in descending order.

In [34]:
%%sql
SELECT province_name, town_name, COUNT(*) AS records_per_town 
FROM location 
GROUP BY province_name, town_name
ORDER BY province_name, records_per_town DESC
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


province_name,town_name,records_per_town
Akatsi,Rural,6290
Akatsi,Lusaka,1070
Akatsi,Harare,800
Akatsi,Kintampo,780
Amanzi,Rural,3100


In [35]:
%%sql
SELECT location_type, COUNT(*) AS num_sources FROM location
GROUP BY location_type
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
2 rows affected.


location_type,num_sources
Urban,15910
Rural,23740


In [36]:
%%sql
SELECT number_of_people_served, COUNT(*) AS people_serveyod FROM water_source 
GROUP BY number_of_people_served 
ORDER BY people_serveyod DESC
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


number_of_people_served,people_serveyod
300,221
302,213
366,212
316,205
390,202


In [37]:
%%sql
SELECT type_of_water_source, COUNT(*) AS num_of_sources FROM water_source 
GROUP BY type_of_water_source
ORDER BY num_of_sources DESC
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


type_of_water_source,num_of_sources
well,17383
tap_in_home,7265
tap_in_home_broken,5856
shared_tap,5767
river,3379


In [38]:
%%sql
SELECT type_of_water_source, CONVERT(AVG(number_of_people_served), DECIMAL(6,0)) AS avg_people_per_source FROM water_source 
GROUP BY type_of_water_source
ORDER BY avg_people_per_source DESC
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


type_of_water_source,avg_people_per_source
shared_tap,2071
river,699
tap_in_home_broken,649
tap_in_home,644
well,279


In [39]:
%%sql
SELECT type_of_water_source, COUNT(*) AS population_served FROM water_source 
GROUP BY type_of_water_source
ORDER BY population_served DESC
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


type_of_water_source,population_served
well,17383
tap_in_home,7265
tap_in_home_broken,5856
shared_tap,5767
river,3379


In [40]:
%%sql
SELECT * FROM water_source 
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


source_id,type_of_water_source,number_of_people_served
AkHa00000224,tap_in_home,956
AkHa00001224,tap_in_home_broken,930
AkHa00002224,tap_in_home_broken,486
AkHa00003224,well,364
AkHa00004224,tap_in_home_broken,942


In [41]:
%%sql
SELECT type_of_water_source, SUM(number_of_people_served) AS population_served FROM water_source 
GROUP BY type_of_water_source
ORDER BY population_served DESC;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


type_of_water_source,population_served
shared_tap,11945272
well,4841724
tap_in_home,4678880
tap_in_home_broken,3799720
river,2362544


In [42]:
%%sql
SELECT SUM(number_of_people_served) AS Total FROM water_source;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
1 rows affected.


Total
27628140


In [43]:
%%sql
SELECT type_of_water_source, ROUND((SUM(number_of_people_served)/27628140 * 100), 0) AS percentage_people_per_source FROM water_source
GROUP BY type_of_water_source
ORDER BY percentage_people_per_source DESC;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


type_of_water_source,percentage_people_per_source
shared_tap,43
well,18
tap_in_home,17
tap_in_home_broken,14
river,9


In [44]:
%%sql
SELECT * FROM water_source LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


source_id,type_of_water_source,number_of_people_served
AkHa00000224,tap_in_home,956
AkHa00001224,tap_in_home_broken,930
AkHa00002224,tap_in_home_broken,486
AkHa00003224,well,364
AkHa00004224,tap_in_home_broken,942


In [45]:
%%sql
SELECT type_of_water_source, SUM(number_of_people_served) AS population_served,
RANK() OVER (ORDER BY SUM(number_of_people_served) DESC) AS rank_by_population
FROM water_source 
GROUP BY type_of_water_source
ORDER BY population_served DESC;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


type_of_water_source,population_served,rank_by_population
shared_tap,11945272,1
well,4841724,2
tap_in_home,4678880,3
tap_in_home_broken,3799720,4
river,2362544,5


In [46]:
%%sql
SELECT * FROM water_source LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


source_id,type_of_water_source,number_of_people_served
AkHa00000224,tap_in_home,956
AkHa00001224,tap_in_home_broken,930
AkHa00002224,tap_in_home_broken,486
AkHa00003224,well,364
AkHa00004224,tap_in_home_broken,942


In [47]:
%%sql
SELECT source_id, type_of_water_source, number_of_people_served, 
DENSE_RANK() OVER (PARTITION BY type_of_water_source ORDER BY number_of_people_served DESC) AS priority_rank 
FROM water_source
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


source_id,type_of_water_source,number_of_people_served,priority_rank
SoRu36791224,river,998,1
SoRu36238224,river,998,1
SoRu36880224,river,998,1
SoRu37756224,river,998,1
SoRu38142224,river,998,1


In [48]:
%%sql
SELECT * FROM visits LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


record_id,location_id,source_id,time_of_record,visit_count,time_in_queue,assigned_employee_id
0,SoIl32582,SoIl32582224,2021-01-01 09:10:00,1,15,12
1,KiRu28935,KiRu28935224,2021-01-01 09:17:00,1,0,46
2,HaRu19752,HaRu19752224,2021-01-01 09:36:00,1,62,40
3,AkLu01628,AkLu01628224,2021-01-01 09:53:00,1,0,1
4,AkRu03357,AkRu03357224,2021-01-01 10:11:00,1,28,14


In [49]:
%%sql
SELECT 
    MAX(CONVERT(time_of_record, DATE)) AS latest_day,
    MIN(CONVERT(time_of_record, DATE)) AS earliest_day,
    DATEDIFF(MAX(CONVERT(time_of_record, DATE)), MIN(CONVERT(time_of_record, DATE))) AS days_difference
FROM visits;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
1 rows affected.


latest_day,earliest_day,days_difference
2023-07-14,2021-01-01,924


In [50]:
%%sql
SELECT AVG(time_in_queue)
FROM visits
WHERE time_in_queue IS NOT NULL AND time_in_queue != 0

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
1 rows affected.


AVG(time_in_queue)
123.2574


In [51]:
%%sql
SELECT DAYNAME(time_of_record) AS day_of_week, ROUND(AVG(time_in_queue), 0) AS avg_queue_time FROM visits 
WHERE time_in_queue IS NOT NULL AND time_in_queue != 0
GROUP BY DAYNAME(time_of_record);

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
7 rows affected.


day_of_week,avg_queue_time
Friday,120
Saturday,246
Sunday,82
Monday,137
Tuesday,108
Wednesday,97
Thursday,105


In [52]:
%%sql
SELECT * FROM visits LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


record_id,location_id,source_id,time_of_record,visit_count,time_in_queue,assigned_employee_id
0,SoIl32582,SoIl32582224,2021-01-01 09:10:00,1,15,12
1,KiRu28935,KiRu28935224,2021-01-01 09:17:00,1,0,46
2,HaRu19752,HaRu19752224,2021-01-01 09:36:00,1,62,40
3,AkLu01628,AkLu01628224,2021-01-01 09:53:00,1,0,1
4,AkRu03357,AkRu03357224,2021-01-01 10:11:00,1,28,14


In [53]:
%%sql
SELECT HOUR(time_of_record) AS hour_of_day, ROUND(AVG(time_in_queue), 0) AS avg_queue_time FROM visits 
GROUP BY HOUR(time_of_record)
ORDER BY HOUR(time_of_record) ASC;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
14 rows affected.


hour_of_day,avg_queue_time
6,149
7,149
8,149
9,49
10,48
11,46
12,47
13,47
14,47
15,48


In [54]:
%%sql
SELECT TIME_FORMAT(TIME(time_of_record), '%H:00') AS hour_of_day, ROUND(AVG(time_in_queue), 0) AS avg_queue_time FROM visits 
GROUP BY TIME_FORMAT(TIME(time_of_record), '%H:00')
ORDER BY TIME_FORMAT(TIME(time_of_record), '%H:00') ASC;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
14 rows affected.


hour_of_day,avg_queue_time
06:00,149
07:00,149
08:00,149
09:00,49
10:00,48
11:00,46
12:00,47
13:00,47
14:00,47
15:00,48


In [55]:
%%sql
SELECT
TIME_FORMAT(TIME(time_of_record), '%H:00') AS hour_of_day, DAYNAME(time_of_record),
CASE
WHEN DAYNAME(time_of_record) = 'Sunday' THEN time_in_queue ELSE NULL END AS Sunday
FROM visits WHERE
time_in_queue != 0
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


hour_of_day,DAYNAME(time_of_record),Sunday
09:00,Friday,None
09:00,Friday,None
10:00,Friday,None
10:00,Friday,None
11:00,Friday,None


In [56]:
%%sql
SELECT
TIME_FORMAT(TIME(time_of_record), '%H:00') AS hour_of_day,
ROUND(AVG(CASE WHEN DAYNAME(time_of_record) = 'Sunday' THEN time_in_queue ELSE NULL END),0) AS Sunday,
ROUND(AVG(CASE WHEN DAYNAME(time_of_record) = 'Monday' THEN time_in_queue ELSE NULL END),0) AS Monday,
ROUND(AVG(CASE WHEN DAYNAME(time_of_record) = 'Tuesday' THEN time_in_queue ELSE NULL END),0) AS Tuesday,
ROUND(AVG(CASE WHEN DAYNAME(time_of_record) = 'Wednesday' THEN time_in_queue ELSE NULL END),0) AS Wednesday,
ROUND(AVG(CASE WHEN DAYNAME(time_of_record) = 'Thursday' THEN time_in_queue ELSE NULL END),0) AS Thursday,
ROUND(AVG(CASE WHEN DAYNAME(time_of_record) = 'Friday' THEN time_in_queue ELSE NULL END),0) AS Friday,
ROUND(AVG(CASE WHEN DAYNAME(time_of_record) = 'Saturday' THEN time_in_queue ELSE NULL END),0) AS Saturday
FROM visits
WHERE
time_in_queue != 0-- this excludes other sources with 0 queue times
GROUP BY hour_of_day
ORDER BY hour_of_day;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
14 rows affected.


hour_of_day,Sunday,Monday,Tuesday,Wednesday,Thursday,Friday,Saturday
06:00,79,190,134,112,134,153,247
07:00,82,186,128,111,139,156,247
08:00,86,183,130,119,129,153,247
09:00,84,127,105,94,99,107,252
10:00,83,119,99,89,95,112,259
11:00,78,115,102,86,99,104,236
12:00,78,115,97,88,96,109,239
13:00,81,122,97,98,101,115,242
14:00,83,127,104,92,96,110,244
15:00,83,126,104,88,92,110,248


# insights
#### 1. Most water sources are rural.
#### 2. 43% of our people are using shared taps. 2000 people often share one tap.
#### 3. 31% of our population has water infrastructure in their homes, but within that group, 45% face non-functional systems due to issues with pipes,
#### pumps, and reservoirs.
#### 4. 18% of our people are using wells of which, but within that, only 28% are clean..
#### 5. Our citizens often face long wait times for water, averaging more than 120 minutes.
#### 6. In terms of queues:
### - Queues are very long on Saturdays.
### - Queues are longer in the mornings and evenings.
### - Wednesdays and Sundays have the shortest queues.

In [57]:
%%sql
SELECT location_id, true_water_source_score 
FROM auditor_report
ORDER BY true_water_source_score DESC
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


location_id,true_water_source_score
AmRu14531,9
AkHa00695,9
AmAb08975,9
SoRu38924,9
KiRu26698,9


#### Joining the auditor_report and visits table

In [58]:
%%sql
SELECT 
  visits.record_id, 
  auditor_report.true_water_source_score, 
  visits.location_id AS visit_location,  
  auditor_report.location_id AS audit_location
FROM 
  auditor_report
JOIN 
  visits
ON 
  auditor_report.location_id = visits.location_id
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


record_id,true_water_source_score,visit_location,audit_location
5185,1,SoRu34980,SoRu34980
59367,3,AkRu08112,AkRu08112
37379,0,AkLu02044,AkLu02044
51627,3,AkHa00421,AkHa00421
28758,0,SoRu35221,SoRu35221


#### Joining auditor_report, visit table and water_quality 

In [59]:
%%sql
SELECT 
  visits.record_id,  
  auditor_report.true_water_source_score AS auditor_score, 
  visits.location_id AS visit_location,  
  auditor_report.location_id AS audit_location,
  water_quality.subjective_quality_score AS survey_score
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality 
ON 
  visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
WHERE 
  auditor_report.true_water_source_score = water_quality.subjective_quality_score
AND visits.visit_count = 1
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


record_id,auditor_score,visit_location,audit_location,survey_score
5185,1,SoRu34980,SoRu34980,1
59367,3,AkRu08112,AkRu08112,3
37379,0,AkLu02044,AkLu02044,0
51627,3,AkHa00421,AkHa00421,3
28758,0,SoRu35221,SoRu35221,0


#### This means out of 1620 auditor_scores 1518 were correct but have limited it to 5 to avoid excess memory consumption

In [60]:
%%sql
SELECT 
  visits.record_id, 
  auditor_report.true_water_source_score AS auditor_score, 
  visits.location_id AS visit_location, 
  auditor_report.location_id AS audit_location,
  water_quality.subjective_quality_score AS survey_score
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality ON visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
WHERE 
  auditor_report.true_water_source_score != water_quality.subjective_quality_score
AND 
  visits.visit_count = 1
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


record_id,auditor_score,visit_location,audit_location,survey_score
21160,3,AkRu05215,AkRu05215,10
7938,3,KiRu29290,KiRu29290,10
43140,9,KiHa22748,KiHa22748,10
18495,6,SoRu37841,SoRu37841,10
33931,1,KiRu27884,KiRu27884,10


#### This means 102 were incorrect but limited it to 5 to prevent memory consumption

In [61]:
%%sql
SELECT 
  visits.record_id, 
  auditor_report.true_water_source_score AS auditor_score, 
  visits.location_id AS visit_location, 
  auditor_report.location_id AS audit_location,
  water_quality.subjective_quality_score AS survey_score,
  water_source.type_of_water_source AS survey_source, 
  auditor_report.type_of_water_source AS auditor_source
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality ON visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
JOIN 
  water_source ON water_source.source_id = visits.source_id 
AND 
  auditor_report.type_of_water_source = water_source.type_of_water_source
WHERE 
  auditor_report.true_water_source_score != water_quality.subjective_quality_score
AND 
  visits.visit_count = 1
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


record_id,auditor_score,visit_location,audit_location,survey_score,survey_source,auditor_source
21160,3,AkRu05215,AkRu05215,10,well,well
7938,3,KiRu29290,KiRu29290,10,shared_tap,shared_tap
43140,9,KiHa22748,KiHa22748,10,tap_in_home_broken,tap_in_home_broken
18495,6,SoRu37841,SoRu37841,10,shared_tap,shared_tap
33931,1,KiRu27884,KiRu27884,10,well,well


In [62]:
%%sql
SELECT 
  visits.record_id, auditor_report.true_water_source_score AS auditor_score, 
  visits.location_id AS visit_location, 
  auditor_report.location_id AS audit_location,
  water_quality.subjective_quality_score AS survey_score,
  auditor_report.type_of_water_source AS auditor_source
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality 
ON 
  visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
WHERE 
  auditor_report.true_water_source_score != water_quality.subjective_quality_score
AND 
  visits.visit_count = 1
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


record_id,auditor_score,visit_location,audit_location,survey_score,auditor_source
21160,3,AkRu05215,AkRu05215,10,well
7938,3,KiRu29290,KiRu29290,10,shared_tap
43140,9,KiHa22748,KiHa22748,10,tap_in_home_broken
18495,6,SoRu37841,SoRu37841,10,shared_tap
33931,1,KiRu27884,KiRu27884,10,well


In [63]:
%%sql
SELECT 
  visits.record_id, 
  auditor_report.true_water_source_score AS auditor_score, 
  visits.location_id AS visit_location, 
  auditor_report.location_id AS audit_location,
  water_quality.subjective_quality_score AS survey_score, 
  auditor_report.type_of_water_source AS auditor_source
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality 
ON 
  visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
WHERE 
  auditor_report.true_water_source_score != water_quality.subjective_quality_score
AND 
  visits.visit_count = 1
LIMIT 5;


 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


record_id,auditor_score,visit_location,audit_location,survey_score,auditor_source
21160,3,AkRu05215,AkRu05215,10,well
7938,3,KiRu29290,KiRu29290,10,shared_tap
43140,9,KiHa22748,KiHa22748,10,tap_in_home_broken
18495,6,SoRu37841,SoRu37841,10,shared_tap
33931,1,KiRu27884,KiRu27884,10,well


In [64]:
%%sql
SELECT 
  employee.assigned_employee_id, 
  visits.record_id,
  visits.location_id,
  auditor_report.true_water_source_score AS auditor_score,
  water_quality.subjective_quality_score AS survey_score
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality 
ON 
  visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
JOIN 
  employee
ON 
  employee.assigned_employee_id = visits.assigned_employee_id
WHERE 
  auditor_report.true_water_source_score != water_quality.subjective_quality_score
AND 
  visits.visit_count = 1
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


assigned_employee_id,record_id,location_id,auditor_score,survey_score
34,21160,AkRu05215,3,10
1,7938,KiRu29290,3,10
1,43140,KiHa22748,9,10
34,18495,SoRu37841,6,10
1,33931,KiRu27884,1,10


In [65]:
%%sql
SELECT 
  employee.employee_name, 
  visits.record_id,
  visits.location_id,
  auditor_report.true_water_source_score AS auditor_score,
  water_quality.subjective_quality_score AS survey_score
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality 
ON 
  visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
JOIN 
  employee
ON 
  employee.assigned_employee_id = visits.assigned_employee_id
WHERE 
  auditor_report.true_water_source_score != water_quality.subjective_quality_score
AND 
  visits.visit_count = 1
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


employee_name,record_id,location_id,auditor_score,survey_score
Rudo Imani,21160,AkRu05215,3,10
Bello Azibo,7938,KiRu29290,3,10
Bello Azibo,43140,KiHa22748,9,10
Rudo Imani,18495,SoRu37841,6,10
Bello Azibo,33931,KiRu27884,1,10


In [66]:
%%sql
WITH Incorrect_record AS  (
SELECT 
  employee.employee_name, 
  visits.record_id,
  visits.location_id,
  auditor_report.true_water_source_score AS auditor_score,
  water_quality.subjective_quality_score AS survey_score
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality 
ON 
  visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
JOIN 
  employee
ON 
  employee.assigned_employee_id = visits.assigned_employee_id
WHERE 
  auditor_report.true_water_source_score != water_quality.subjective_quality_score
AND 
  visits.visit_count = 1
)
SELECT DISTINCT(employee_name) FROM Incorrect_record;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
17 rows affected.


employee_name
Rudo Imani
Bello Azibo
Zuriel Matembo
Yewande Ebele
Jengo Tumaini
Farai Nia
Malachi Mavuso
Makena Thabo
Lalitha Kaburi
Gamba Shani


In [67]:
%%sql
WITH Incorrect_record AS  (
SELECT 
  employee.employee_name, 
  visits.record_id,
  visits.location_id,
  auditor_report.true_water_source_score AS auditor_score,
  water_quality.subjective_quality_score AS survey_score
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality 
ON 
  visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
JOIN 
  employee
ON 
  employee.assigned_employee_id = visits.assigned_employee_id
WHERE 
  auditor_report.true_water_source_score != water_quality.subjective_quality_score
AND 
  visits.visit_count = 1
)
SELECT employee_name, COUNT(*) AS Number_of_mistake FROM Incorrect_record
GROUP BY employee_name
ORDER BY Number_of_mistake DESC
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


employee_name,Number_of_mistake
Bello Azibo,26
Malachi Mavuso,21
Zuriel Matembo,17
Lalitha Kaburi,7
Rudo Imani,5


In [68]:
%%sql
CREATE VIEW Incorrect_record AS (
SELECT 
  employee.employee_name, 
  visits.record_id,
  visits.location_id,
  auditor_report.true_water_source_score AS auditor_score,
  water_quality.subjective_quality_score AS survey_score
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality 
ON 
  visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
JOIN 
  employee
ON 
  employee.assigned_employee_id = visits.assigned_employee_id
WHERE 
  auditor_report.true_water_source_score != water_quality.subjective_quality_score
AND 
  visits.visit_count = 1
)

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
(pymysql.err.OperationalError) (1050, "Table 'Incorrect_record' already exists")
[SQL: CREATE VIEW Incorrect_record AS (
SELECT 
  employee.employee_name, 
  visits.record_id,
  visits.location_id,
  auditor_report.true_water_source_score AS auditor_score,
  water_quality.subjective_quality_score AS survey_score
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality 
ON 
  visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
JOIN 
  employee
ON 
  employee.assigned_employee_id = visits.assigned_employee_id
WHERE 
  auditor_report.true_water_source_score != water_quality.subjective_quality_score
AND 
  visits.visit_count = 1
)]
(Background on this error at: https://sqlalche.me/e/20/e3q8)


In [69]:
%%sql
SELECT 
  * 
FROM 
  md_water_services.incorrect_record
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


employee_name,record_id,location_id,auditor_score,survey_score
Rudo Imani,21160,AkRu05215,3,10
Bello Azibo,7938,KiRu29290,3,10
Bello Azibo,43140,KiHa22748,9,10
Rudo Imani,18495,SoRu37841,6,10
Bello Azibo,33931,KiRu27884,1,10


In [70]:
%%sql
WITH error_count AS (
SELECT
  employee_name,
  COUNT(*) AS number_of_mistakes
FROM
  md_water_services.incorrect_record
GROUP BY 
  employee_name
ORDER BY 
  number_of_mistakes DESC)
SELECT * FROM error_count
LIMIT 5;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
5 rows affected.


employee_name,number_of_mistakes
Bello Azibo,26
Malachi Mavuso,21
Zuriel Matembo,17
Lalitha Kaburi,7
Rudo Imani,5


In [71]:
%%sql
WITH error_count AS (
SELECT
  employee_name,
  COUNT(*) AS number_of_mistakes
FROM
  md_water_services.incorrect_record
GROUP BY 
  employee_name
ORDER BY 
  number_of_mistakes DESC)
SELECT AVG(number_of_mistakes) AS average_mistake FROM error_count;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
1 rows affected.


average_mistake
6.0000


In [72]:
%%sql
SELECT 
    employee_name,
    COUNT(*) AS number_of_mistakes
FROM 
    md_water_services.incorrect_record
GROUP BY 
    employee_name
HAVING 
    COUNT(*) > (
        SELECT AVG(mistake_count)
        FROM (
            SELECT COUNT(*) AS mistake_count
            FROM md_water_services.incorrect_record
            GROUP BY employee_name
        ) AS subquery
    )
ORDER BY 
    number_of_mistakes DESC;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
4 rows affected.


employee_name,number_of_mistakes
Bello Azibo,26
Malachi Mavuso,21
Zuriel Matembo,17
Lalitha Kaburi,7


In [90]:
%%sql
CREATE OR REPLACE VIEW md_water_services.suspect_list AS
SELECT 
    employee_name,
    COUNT(*) AS number_of_mistakes
FROM 
    md_water_services.incorrect_record
GROUP BY 
    employee_name
HAVING 
    COUNT(*) > (
        SELECT AVG(mistake_count)
        FROM (
            SELECT COUNT(*) AS mistake_count
            FROM md_water_services.incorrect_record
            GROUP BY employee_name
        ) AS subquery
    )
ORDER BY 
    number_of_mistakes DESC;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
0 rows affected.


[]

In [93]:
%%sql
SELECT * FROM md_water_services.suspect_list;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
4 rows affected.


employee_name,number_of_mistakes
Bello Azibo,26
Malachi Mavuso,21
Zuriel Matembo,17
Lalitha Kaburi,7


#### I have added another column auditor_report.statements

In [80]:
%%sql
CREATE OR REPLACE VIEW md_water_services.incorrect_record AS(
SELECT 
  employee.employee_name, 
  visits.record_id,
  visits.location_id,
  auditor_report.true_water_source_score AS auditor_score,
  water_quality.subjective_quality_score AS survey_score,
  auditor_report.statements
FROM 
  auditor_report
JOIN 
  visits 
ON 
  auditor_report.location_id = visits.location_id
JOIN 
  water_quality 
ON 
  visits.visit_count = water_quality.visit_count 
AND 
  visits.record_id = water_quality.record_id
JOIN 
  employee
ON 
  employee.assigned_employee_id = visits.assigned_employee_id
WHERE 
  auditor_report.true_water_source_score != water_quality.subjective_quality_score
AND 
  visits.visit_count = 1
)


 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
0 rows affected.


[]

In [91]:
%%sql
SELECT
  employee_name, 
  location_id, 
  statements 
FROM
  md_water_services.incorrect_record 
WHERE 
 md_water_services.incorrect_record.employee_name IN (
    SELECT employee_name
    FROM md_water_services.suspect_list);

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
71 rows affected.


employee_name,location_id,statements
Bello Azibo,KiRu29290,"A young artist sketches the faces in the queue, capturing the weariness of daily hours spent waiting for water."
Bello Azibo,KiHa22748,"A young girl's hopeful eyes are clouded by mistrust, her innocence tarnished by the corrupt system."
Bello Azibo,KiRu27884,"A traditional healer's empathy turns to bitterness, knowing that corrupt practices harm her community."
Zuriel Matembo,KiZu31170,"A community leader stood with his people, expressing concern for the water quality and the time lost in queues."","""
Bello Azibo,AkRu06495,"A healthcare worker in the queue expressed fears about water-borne diseases, her face etched with worry."","""
Zuriel Matembo,SoRu38331,"An unsettling atmosphere surrounded the official, as villagers shared their experiences of arrogance and lack of dedication. The mention of cash exchanges only intensified their doubts."
Malachi Mavuso,AmAm09607,Villagers spoke of an unsettling encounter with an official who appeared dismissive and detached. The reference to cash transactions added to their growing sense of distrust.
Zuriel Matembo,AkHa00314,"A street vendor's sales suffer from time spent waiting, her concern for the water's quality affecting her products."
Malachi Mavuso,KiRu26598,"A teenager's dreams are tempered by reality, her future threatened by the corrupt practices she sees around her."
Bello Azibo,KiIs23853,Villagers' wary accounts of an official's arrogance and detachment from their concerns raised suspicions. The mention of cash changing hands further tainted their perception.


In [92]:
%%sql
SELECT
  employee_name, 
  location_id, 
  statements 
FROM
  md_water_services.incorrect_record 
WHERE 
 md_water_services.incorrect_record.employee_name IN (
    SELECT employee_name
    FROM md_water_services.suspect_list)
AND statements LIKE '%cash%';

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
19 rows affected.


employee_name,location_id,statements
Zuriel Matembo,SoRu38331,"An unsettling atmosphere surrounded the official, as villagers shared their experiences of arrogance and lack of dedication. The mention of cash exchanges only intensified their doubts."
Malachi Mavuso,AmAm09607,Villagers spoke of an unsettling encounter with an official who appeared dismissive and detached. The reference to cash transactions added to their growing sense of distrust.
Bello Azibo,KiIs23853,Villagers' wary accounts of an official's arrogance and detachment from their concerns raised suspicions. The mention of cash changing hands further tainted their perception.
Bello Azibo,HaSe21323,Villagers spoke of an unsettling encounter with an official who appeared dismissive and detached. The reference to cash transactions added to their growing sense of distrust.
Zuriel Matembo,AkRu05880,Villagers' wary accounts of an official's arrogance and detachment from their concerns raised suspicions. The allusion to cash changing hands deepened their skepticism.
Bello Azibo,KiRu27065,Villagers expressed their discomfort with an official who displayed a haughty demeanor and negligence. The mention of cash transactions deepened their growing sense of unease.
Malachi Mavuso,KiRu25347,Villagers expressed their discontent with an official who appeared dismissive and neglectful. The mention of cash changing hands added to their growing sense of distrust.
Zuriel Matembo,SoIl32575,Villagers recounted unsettling encounters with an official known for their arrogance and avoidance of responsibilities. The mention of cash changing hands added to their apprehension and distrust.
Bello Azibo,AkRu04508,"An unsettling atmosphere surrounded the official, as villagers shared their experiences of arrogance and lack of dedication. The mention of cash exchanges only intensified their doubts."
Lalitha Kaburi,AkRu07310,"Villagers spoke of their unsettling encounters with an official who seemed indifferent and uninterested, hinting at potential improprieties involving cash exchanges."


In [94]:
%%sql
SELECT
  employee_name, 
  location_id, 
  statements 
FROM
  md_water_services.incorrect_record 
WHERE 
 md_water_services.incorrect_record.employee_name IN (
    SELECT employee_name
    FROM md_water_services.suspect_list)
AND statements LIKE '%cash%'
AND md_water_services.incorrect_record.employee_name != md_water_services.incorrect_record.employee_name;

 * mysql+pymysql://root:***@127.0.0.1:3306/md_water_services
0 rows affected.


employee_name,location_id,statements
